### ISSUE TITLE
Erroneous warning for removing the space after = when assign empty strings.

### ISSUE THREAD

**Issue by orbea** [koalaman/shellcheck#1001]
#### For bugs
- Rule Id: SC1007
- My shellcheck version: `online`
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

set -eu

a= b= c=

echo "$a $b $c"

```

#### Here's what shellcheck currently says:

```
Line 5:
a= b= c=
 ^-- SC1007: Remove space after = if trying to assign a value (for empty string, use var='' ... ).
    ^-- SC1007: Remove space after = if trying to assign a value (for empty string, use var='' ... ).
```

#### Here's what I wanted or expected to see:

Nothing.

Rule `SC1007` discusses setting incorrectly setting environment of script variables, but the given example only shows setting several blank variables in a single line the given rationale is not relevant. This can be avoided with `a=; b=; c=`, but that is not required for correct behavior in a script. Explicitly setting blank variables like this can be useful if you plan to loop over something like `foo="${foo} $bar"`.

**Comment by koalaman** [comment]
The shell's behavior is only correct if you expect it. If you change the variable names in your example, the intended behavior is more ambiguous:

    config= b64password= Ym9pbms=

The warning suggests using `a='' b='' c=''` if you intend to assign empty strings, not to correct the behavior but to make your intention explicit.

You can consider doing it this way to help out any casuals reading the code, since 95%* of the time they do not expect a space after the `=` to matter. It's always annoying to be warned about correct code, but I believe that overall, this warning is more helpful to noobs than it is annoying to experts.

\* I counted, N=19

**Comment by orbea** [comment]
Fair enough, thanks for the response.
